src/redist/targets/packaging/windows/dotnetCoreUninstall/dotnet.wxs

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+  <?include "Variables.wxi" ?>
+  <Product Id="*" Name="$(var.ProductName)" Language="$(var.ProductLanguage)" Version="$(var.ProductVersion)" Manufacturer="$(var.Manufacturer)" UpgradeCode="$(var.UpgradeCode)">
+    <Package Compressed="yes" InstallScope="perMachine" InstallerVersion="200" />
+
+    <MajorUpgrade DowngradeErrorMessage="$(var.DowngradeErrorMessage)" Schedule="afterInstallInitialize"/>
+
+    <?if $(var.Platform)=x86?>
+    <MediaTemplate EmbedCab="yes" />
+    <?elseif $(var.Platform)=x64?>
+    <MediaTemplate EmbedCab="yes" />
+    <?endif?>
+
+    <Feature Id="MainFeature" Title="Main Feature" Level="1">
+      <ComponentGroupRef Id="InstallFiles" />
+      <ComponentGroupRef Id="InstallSharedHostandDetectionKeys" />
+    </Feature>
+    <Feature Id="Provider" Absent="disallow" AllowAdvertise="no" Description="Used for Ref Counting" Display="hidden" Level="1" InstallDefault="local" Title="RefCounting" TypicalDefault="install">
+      <ComponentRef Id="$(var.DependencyKeyId)" />
+    </Feature>
+    <Property Id="ProductFamily" Value="$(var.ProductFamily)" />
+    <Property Id="ProductEdition" Value="$(var.ProductEdition)" />
+    <Property Id="ProductCPU" Value="$(var.Platform)" />
+    <Property Id="RTM_ProductVersion" Value="$(var.Dotnet_ProductVersion)" />
+    <Property Id="MSIFASTINSTALL" Value="7" />
+    <WixVariable Id="WixUILicenseRtf" Value="$(var.MicrosoftEula)" />
+
+    <Property Id="DOTNETCOREUNINSTALLTEXE">
+      <DirectorySearch Id="DOTNETPATH" Path="[#DOTNETDOTNETCOREUNINSTALLHOME]">
```

Comment:
How are you setting the install directory if the 'InstallSharedHostandDetectionKeys' fails to find a host? Or do you plan to error the install and quit?

Reply:
Also why is it important to install where the ShardHost is? Can't we just install to ProgramFiles?

Reply:
ah, i get it. I could just remove this part. I didn't really understand what these code was doing

Reply:
Removed

Reply:
You should use:
ProgramFilesFolder - gets the well-known folder for CSIDL_PROGRAM_FILESX86.
ProgramFiles64Folder - gets the well-known folder for CSIDL_PROGRAM_FILES.
for the installation directory.

Something like this:
```
<?if $(var.Platform)=x86?>
DOTNETDOTNETCOREUNINSTALLHOME = ProgramFilesFolder
<?elseif $(var.Platform)=x64?>
DOTNETDOTNETCOREUNINSTALLHOME = ProgramFiles64Folder
<?endif?>
```

Reply:
@joeloff 
Can you review this PR?